curs_lib.c

```diff
@@ -543,28 +531,44 @@ void mutt_flush_stdin(void)
  */
 int mutt_any_key_to_continue(const char *s)
 {
-  struct termios t;
+  struct termios term;
   struct termios old;
 
   int fd = open("/dev/tty", O_RDONLY);
   if (fd < 0)
     return EOF;
-  tcgetattr(fd, &t);
-  memcpy((void *) &old, (void *) &t, sizeof(struct termios)); /* save original state */
-  t.c_lflag &= ~(ICANON | ECHO);
-  t.c_cc[VMIN] = 1;
-  t.c_cc[VTIME] = 0;
-  tcsetattr(fd, TCSADRAIN, &t);
-  fflush(stdout);
+
+  tcgetattr(fd, &old); // Save the current tty settings
+
+  term = old;
+  term.c_lflag &= ~(ICANON | ECHO); // Canonical (not line-buffered), don't echo the characters
+  term.c_cc[VMIN] = 1;    // Wait for at least one character
+  term.c_cc[VTIME] = 255; // Wait for 25.5s
+  tcsetattr(fd, TCSANOW, &term);
+
   if (s)
     fputs(s, stdout);
   else
     fputs(_("Press any key to continue..."), stdout);
   fflush(stdout);
-  int ch = fgetc(stdin);
-  mutt_flush_stdin();
-  tcsetattr(fd, TCSADRAIN, &old);
+
+  char ch = '\0';
+  // Wait for a character.  This might timeout, so loop.
+  while (read(fd, &ch, 1) == 0)
```

Comment:
Is there a more readable way to construct this control flow? I missed the `;` the first time I read this code.

Reply:
A `;` on a blank line is common, idiomatic, C.
I count over 40 occurrences in the code.

Reply:
I trust your judgment since you're much more experience with C than me.

Reply:
@flatcap how did you count blank lines?
This would count too

````
while (read(fd, &ch, 1) == 0)
{
    ;
}
````


I missed the `;` on a first read too.
Personally I would prefer braces even for such simple statements (granted, with neomutt style guide policy about braces it doubles the line numbers), and it also makes code more consistent.
But I know some people prefer it this way, as stated in neomutt coding style guide..

Reply:
> how did you count blank lines?

Nothing fancy: `grep -B1 "^ \+;$" -R .`

> I would prefer braces even for such simple statements

An alternative would be a comment:

```c
while (condition)
  ; // Do nothing
```

I'll give it some consideration.

> neomutt coding style guide

I tried to keep it short, so people would read it.
At some point, I'd like to write down every 'rule' I've consciously applied.